samples/subsys/mgmt/mcumgr/smp_svr/README.rst

```diff
@@ -1,117 +1,271 @@
 
 Building and Running
 ********************
 
-The below steps describe how to build and run the `smp_svr` sample app in
-Zephyr.  Where examples are given, they assume the following setup:
-
-* BSP: Nordic nRF52dk
-* MCU: PCA10040
+The below steps describe how to build and run the ``smp_svr`` sample in
+Zephyr. Where examples are given, they assume the sample is being built for
+the Nordic nRF52 Development Kit (``BOARD=nrf52_pca10040``).
 
-If you would like to use a more constrained platform, such as the nRF51, you
-should use the `prj.conf.tiny` configuration file rather than the default
-`prj.conf`.
+If you would like to use a more constrained platform, such as the nRF51 DK, you
+should use the :file:`prj.conf.tiny` configuration file rather than the default
+:file:`prj.conf`.
 
 Step 1: Build MCUboot
 =====================
 
-Build MCUboot by following the instructions in its `docs/readme-zephyr.md`
-file.
+Build MCUboot by following the instructions in the :ref:`mcuboot`
+documentation page.
 
-Step 2: Upload MCUboot
+Step 2: Flash MCUboot
 ======================
 
-Upload the resulting `zephyr.bin` file to address 0 of your board.  This can be
-done in gdb as follows:
+Flash the resulting image file to address 0x0 of flash memory.
+This can be done in multiple ways.
 
-```
-restore <path-to-mcuboot-zephyr.bin> binary 0
-```
+Using make or ninja:
+
+.. code-block:: console
+
+   make flash
+   # or
+   ninja flash
+
+Using GDB:
+
+.. code-block:: console
+
+   restore <path-to-mcuboot-zephyr.bin> binary 0
 
 Step 3: Build smp_svr
 =====================
 
-`smp_svr` can be built for the nRF52 as follows:
+``smp_svr`` can be built for the nRF52 as follows:
 
 .. zephyr-app-commands::
     :zephyr-app: samples/mgmt/mcumgr/smp_svr
     :board: nrf52_pca10040
     :build-dir: nrf52_pca10040
     :goals: build
 
-Step 4: Create an MCUboot-compatible image
-==========================================
+.. _smp_svr_sample_sign:
+
+Step 4: Sign the image
+======================
+
+.. note::
+   From this section onwards you can use either a binary (``.bin``) or an
+   Intel Hex (``.hex``) image format. This is written as ``(bin|hex)`` in this
+   document.
 
-Using MCUboot's `imgtool.py` script, convert the `zephyr.bin` file from step 3
-into an image file.  In the below example, the MCUboot repo is located at
-`~/repos/mcuboot`.
+Using MCUboot's :file:`imgtool.py` script, sign the :file:`zephyr.(bin|hex)`
+file you built in Step 3. In the below example, the MCUboot repo is located at
+:file:`~/src/mcuboot`.
 
 .. code-block:: console
 
-    ~/repos/mcuboot/scripts/imgtool.py sign \
+   ~/src/mcuboot/scripts/imgtool.py sign \
+        --key ~/src/mcuboot/root-rsa-2048.pem \
         --header-size 0x200 \
         --align 8 \
         --version 1.0 \
         --included-header \
-        --key ~/repos/mcuboot/root-rsa-2048.pem \
-        <path-to-smp_svr-zephyr.bin> signed.img
+        <path-to-zephyr.(bin|hex)> signed.(bin|hex)
```

Comment:
Intel hex is not supported by McuMgr GO application (while imgtool is able to produce image in such format). Maybe this should be pointed somewhere.

Reply:
got it - you had mentioned this in FWU guide section.